pi4j-test/src/test/java/com/pi4j/test/smokeTest/provider/ProviderSmokeTest.java

```diff
@@ -0,0 +1,264 @@
+        pi4j = Pi4J.newAutoContext();
+
+    }
+
+
+    @AfterAll
+    public static void afterTest() {
+        try {
+            pi4j.shutdown();
+        } catch (Pi4JException e) { /* do nothing */ }
+    }
+
+
+    @Test
+    public void testI2c() {
+        final int idValueMskBMP = 0x58;   // expected chpId value BMP280
+        final int idValueMskBME = 0x60;   // expected chpId value BME280
+
+        I2C dev = createI2cBMPDevice();
+        // read 0xD0 validate data equal 0x58 or 0x60
+        int id = dev.readRegister(0xD0);
+
+        Assertions.assertTrue(id == idValueMskBMP || id ==idValueMskBME, "TestBMP and BME ID");
+    }
+
+    @Test
+    public void testSpi() {
+        final int idValueMskBMP = 0x58;   // expected chpId value BMP280
+        final int idValueMskBME = 0x60;   // expected chpId value BME280
+
+        Spi spi = createSPIDevice();
+        final int chipId = 0xD0;
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        int id = this.readSpiRegister(spi, chipId);
+
+        Assertions.assertTrue(id == idValueMskBMP || id ==idValueMskBME, "TestBMP and BME ID");
+    }
+
+    @Test
+    public void testPWM() throws Exception {
+        final int PWM_CHANNEL = 2;
+        final int PIN_GPIO23 = 23;
+        pwmFlashes = 0 ;
+        DigitalInput gpio23InMonitor = createDigitalInput(PIN_GPIO23);
+        gpio23InMonitor.addListener(new ProviderSmokeTest.DataInGpioListener());
+        Pwm pwm = createPWM(PWM_CHANNEL);
+        pwm.on(50, 1) ;
+        Thread.sleep(10000);  // wait 10 seconds while listener counts flashes
+
+        Assertions.assertTrue(pwmFlashes == 10, "TestPWM flashing 10 times in 10 seconds");
+    }
+
+    @Test
+    public void testGpioIn() {
+          final int PIN_GPIO16 = 16;
+        final int PIN_GPIO26 = 26;
+        DigitalOutput gpio26OutControl = createDigitalOutput(PIN_GPIO26);
+        DigitalInput gpio16InTest = createDigitalInput(PIN_GPIO16);
+        // Validate monitor is LOW, then set test control state HIGH
+        if (gpio16InTest.state() == DigitalState.LOW) {
+            gpio26OutControl.high();
+        }
+        DigitalState state = gpio16InTest.state();
+
+        Assertions.assertTrue(state  ==  DigitalState.HIGH, "TestGpioIn  now HIGHs");
+    }
+
+    @Test
+    public void testGpioOut() {
+        final int PIN_GPIO24 = 24;
+        final int PIN_GPIO25 = 25;
+
+        DigitalOutput gpio24OutTest = createDigitalOutput(PIN_GPIO24);
+
+        DigitalInput gpio25InMonitor = createDigitalInput(PIN_GPIO25);
+        // Validate monitor is LOW, test control state HIGH
+        if (gpio25InMonitor.state() == DigitalState.LOW) {
+            gpio24OutTest.high();
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
The sleep is here while i debug ffm. Without the pause the test usually fails. When i have enough details i will open an issue to s needed